imdssample.java

```diff
@@ -1,40 +1,172 @@
+import org.bouncycastle.util.encoders.Base64;
+import com.google.gson.Gson;
 
-   public static String sendGet() throws Exception {
+/**
+ * This example has two dependencies:
+ * 	1. Google Gson library: https://github.com/google/gson
+ * 	2. BouncyCastle Java library: https://www.bouncycastle.org/java.html
+ */
+public class IMDSSample 
+{
+    public static final String ImdsServer = "http://169.254.169.254";
+    public static final String InstanceEndpoint = ImdsServer + "/metadata/instance";
+    public static final String AttestedEndpoint = ImdsServer + "/metadata/attested/document";
+    public static final String NonceValue = "123456";
+    
+	public static void main(String[] args)
+	{
+		// Query /instance metadata
+		String result = QueryInstanceEndpoint();
+        ParseInstanceResponse(result);
+
+        // Make Attested call and parse the response
+        result = QueryAttestedEndpoint();
+        ParseAttestedResponse(result);
+	}
+	
+	private static void ParseInstanceResponse(String response)
+	{
+		System.out.println("Instance response: " + response);
+	}
+	
+	private static void ParseAttestedResponse(String response)
+	{
+		System.out.println("Parsing Attested response");
+		AttestedDocument document = new Gson().fromJson(response, AttestedDocument.class);
+		byte[] decoded = Base64.decode(document.signature);
+		ValidateCertificate(decoded);
+		ValidateAttestedData(decoded);
+	}
+	
+	private static void ValidateCertificate(byte[] decoded)
+	{
+		try
+		{
+			CertificateFactory factory =  CertificateFactory.getInstance("X.509");
+			Collection certs = factory.generateCertificates(new ByteArrayInputStream(decoded));
+			Iterator it = certs.iterator();
+			while(it.hasNext())
+			{
+				X509Certificate cert = (X509Certificate)it.next();
+				cert.checkValidity();
+				X500Principal issuer = cert.getIssuerX500Principal();
+				System.out.println("Issuer: " + issuer.toString());
+				X500Principal subject = cert.getSubjectX500Principal();
+				System.out.println("Subject: " + subject.toString());
+				System.out.println("Valid until: " + cert.getNotAfter().toString());
+			}
+		}
+		catch(Exception ex)
+		{
+			System.out.println("Exception validating certificate: " + ex.getMessage());
+		}
+	}
+	
+	private static void ValidateAttestedData(byte[] decoded)
+	{
+		try
+		{
+			CMSSignedData signature = new CMSSignedData(decoded);
+			CMSTypedData signedData = signature.getSignedContent();
+			String signedDataString = new String((byte[])signedData.getContent());
+			System.out.println("Attested data: " + signedDataString);
+			AttestedData data = new Gson().fromJson(signedDataString, AttestedData.class);
+			if(data.nonce.compareTo(NonceValue) == 0)
+			{
+				System.out.println("Nonce values match");
+			}
+		}
+		catch(Exception ex)
+		{
+			System.out.println("Exception validating data: " + ex.getMessage());
+		}
+	}
+	
+	private static String QueryInstanceEndpoint()
+	{
+		return QueryImds(InstanceEndpoint, "2017-04-02");		
```

Comment:
since we are already updating the example ,should we update the api-version  for instance too ?

Reply:
Done. We also probably should update this in the other examples, as well. I'll do that in a separate PR.